server/monitor-types/dns.js

```diff
@@ -17,36 +27,56 @@ class DnsMonitorType extends MonitorType {
         let dnsRes = await dnsResolve(monitor.hostname, monitor.dns_resolve_server, monitor.port, monitor.dns_resolve_type);
         heartbeat.ping = dayjs().valueOf() - startTime;
 
-        if (monitor.dns_resolve_type === "A" || monitor.dns_resolve_type === "AAAA" || monitor.dns_resolve_type === "TXT" || monitor.dns_resolve_type === "PTR") {
-            dnsMessage += "Records: ";
-            dnsMessage += dnsRes.join(" | ");
-        } else if (monitor.dns_resolve_type === "CNAME" || monitor.dns_resolve_type === "PTR") {
-            dnsMessage += dnsRes[0];
-        } else if (monitor.dns_resolve_type === "CAA") {
-            dnsMessage += dnsRes[0].issue;
-        } else if (monitor.dns_resolve_type === "MX") {
-            dnsRes.forEach(record => {
-                dnsMessage += `Hostname: ${record.exchange} - Priority: ${record.priority} | `;
-            });
-            dnsMessage = dnsMessage.slice(0, -2);
-        } else if (monitor.dns_resolve_type === "NS") {
-            dnsMessage += "Servers: ";
-            dnsMessage += dnsRes.join(" | ");
-        } else if (monitor.dns_resolve_type === "SOA") {
-            dnsMessage += `NS-Name: ${dnsRes.nsname} | Hostmaster: ${dnsRes.hostmaster} | Serial: ${dnsRes.serial} | Refresh: ${dnsRes.refresh} | Retry: ${dnsRes.retry} | Expire: ${dnsRes.expire} | MinTTL: ${dnsRes.minttl}`;
-        } else if (monitor.dns_resolve_type === "SRV") {
-            dnsRes.forEach(record => {
-                dnsMessage += `Name: ${record.name} | Port: ${record.port} | Priority: ${record.priority} | Weight: ${record.weight} | `;
-            });
-            dnsMessage = dnsMessage.slice(0, -2);
+        const conditions = ConditionExpressionGroup.fromMonitor(monitor);
+        let conditionsResult = null;
+        const handleConditions = (data) => conditions ? evaluateExpressionGroup(conditions, data) : null;
```

Comment:
Follow-up to https://github.com/louislam/uptime-kuma/pull/5048#discussion_r1730408556

Can `conditions` ever be falsy?

I don't think so, right? => Am I missing something?